### PR TITLE
RDCC-2483: Make primary base location field mandatory

### PR DESCRIPTION
**Pull request by ishratt**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-2483

### Change description ###
RDCC-2483: Make primary base location field mandatory

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by nitishsingla1234 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/util/CaseWorkerConstants.java [hunk 1]
```diff
@@ -99,7 +99,8 @@ private CaseWorkerConstants() {
     public static final String FILE_MISSING_HEADERS = "The file is missing some column headers."
         + " Review and upload again";
 
-    public static final String NO_LOCATION_PRESENT = "You must add location details and upload the file again.";
+    public static final String NO_PRIMARY_LOCATION_PRESENT = "You must add Primary Base Location Name and"
```
Comment (SujeethKumarAri) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/util/CaseWorkerConstants.java:102]: There should be space after `and`

**Review by SujeethKumarAri — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/util/CaseWorkerConstants.java (hunk 1 above)
Comment (ishratt) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/util/CaseWorkerConstants.java:102]: ah I missed this! thanks @SujeethKumarAri will amend it now

File: src/main/java/uk/gov/hmcts/reform/cwrdapi/util/CaseWorkerChildListValidator.java [hunk 2]
```diff
@@ -58,6 +60,14 @@ private boolean isValidLocations(CaseWorkerProfile caseWorkerProfile, Constraint
                     .addPropertyNode(LOCATION_FIELD)
                     .addConstraintViolation();
             }
+        } else if (isEmpty(caseWorkerProfile.getLocations())
+                   || (caseWorkerProfile.getLocations().size() == 1
```
Comment (SujeethKumarAri) [inline src/main/java/uk/gov/hmcts/reform/cwrdapi/util/CaseWorkerChildListValidator.java:64]: Sorry for not mentioning this before, I think `caseWorkerProfile.getLocations().size() == 1` is not required here as size > 1 check is being done in the first if condition 
`caseWorkerProfile.getLocations().size() > 1`

Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/util/CaseWorkerChildListValidator.java:64]: @SujeethKumarAri  I think this is fine we have 2 locations hence >1  means we must get primary. hence --1 should be fine.

Comment (SujeethKumarAri) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/util/CaseWorkerChildListValidator.java:64]: @abhijitdiwan - we have two locations and we check > 1 in the first` if`, so it is not mandatory to check if it is ==1 in next `else if `

Comment (ishratt) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/util/CaseWorkerChildListValidator.java:64]: Yes I think @SujeethKumarAri is right, the size == 1 check is redundant at this point, as we know that if it's not empty then the size is definitely 1. Thanks both.

Comment (abhijitdiwan) [reply src/main/java/uk/gov/hmcts/reform/cwrdapi/util/CaseWorkerChildListValidator.java:64]: Yes correct...We should use empty and last condition with stream to check if primary present or not hence in future we got more locations it won't need change

**Review by abhijitdiwan — APPROVED** [review]

**Review by SujeethKumarAri — APPROVED** [review]

**Pull request merged**